atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/ranges/DbKvsGetRanges.java

```diff
@@ -141,22 +138,27 @@ public DbKvsGetRanges(
     }
 
     private static SortedSetMultimap<Integer, byte[]> getRowsForBatches(
-            SqlConnection connection,
+            Supplier<SqlConnection> connectionSupplier,
```

Comment:
copying from gbonik: 

I don't understand what exactly you suggest. SqlConnectionSupplier still returns a SqlConnection which is not AutoCloseable, and moreover is owned by the supplier and is not supposed to be closed.

We could write something like this?

try (SqlConnectionSupplier supplier = new ConnectionSupplier(connectionSupplier)) {
   // use supplier.get()
}
Honestly, I can't see how this code is better. Just too many different suppliers.